File: BACK-END/KuentecoChat/src/main/java/com/example/back_end/dto/response/StringChatResponseDTO.java
package com.example.back_end.dto.response;

import com.example.back_end.dto.response.ai.ChartDataResponseDTO;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class StringChatResponseDTO {
    private String conversationId;
    private String response;
    private ChartDataResponseDTO chartData;
}

File: FRONT-END/lib/dto/chat/response/string_chat_response_dto.dart
import 'chart_data_response_dto.dart';

class StringChatResponseDTO {
  final String conversationId;
  final String response;
  final ChartDataResponseDTO? chartData;

  StringChatResponseDTO({
    required this.conversationId,
    required this.response,
    this.chartData,
  });

  factory StringChatResponseDTO.fromJson(Map<String, dynamic> json) {
    return StringChatResponseDTO(
      conversationId: json['conversationId'] ?? '',
      response: json['response'] ?? '',
      chartData: json['chartData'] != null 
          ? ChartDataResponseDTO.fromJson(json['chartData'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'conversationId': conversationId,
      'response': response,
      if (chartData != null) 'chartData': chartData!.toJson(),
    };
  }
}